builder/swig/go.bzl

```diff
@@ -0,0 +1,174 @@
+        outputs = [wrap_src, wrap_go_dir] + swig_headers,
+        executable = ctx.file._swig,
+        arguments = args,
+    )
+
+    lib_compilation_context = ctx.attr.lib[CcInfo].compilation_context
+    compilation_context = cc_common.create_compilation_context(
+        headers = depset(swig_headers, transitive = [lib_compilation_context.headers]),
+        defines = lib_compilation_context.defines,
+        framework_includes = lib_compilation_context.framework_includes,
+        includes = depset(
+            [h.dirname for h in ctx.attr.lib[CcInfo].compilation_context.headers.to_list()],
+            transitive = [lib_compilation_context.includes],
+        ),
+        local_defines = lib_compilation_context.local_defines,
+        quote_includes = lib_compilation_context.quote_includes,
+        system_includes = depset(
+            [],
+            transitive = [lib_compilation_context.system_includes],
+        )
+    )
+
+    return [
+        DefaultInfo(files = depset([wrap_src])),
+        OutputGroupInfo(
+            go_source = depset([wrap_go_dir]),
+            cxx_source = depset([wrap_src]),
+        ),
+        CcInfo(
+            compilation_context = compilation_context,
+            linking_context = ctx.attr.lib[CcInfo].linking_context,
+        ),
+    ]
+
+_swig_go_wrapper = rule(
+    implementation = _swig_go_wrapper_impl,
+    attrs = {
+        "lib": attr.label(
+            doc = "The C library for which to generate the wrapper",
+            providers = [CcInfo],
+            mandatory = True,
+        ),
+        "interface": attr.label(
+            doc = "Optional SWIG interface (.i) file",
+            allow_single_file = True,
+        ),
+        "includes": attr.label_list(
+            doc = "Additional SWIG files required for wrapper generation",
+            allow_files = True,
+        ),
+        "package_name": attr.string(
+            doc = "Package name override for generated Go package",
+        ),
+        "enable_cxx": attr.bool(
+            doc = "Enable SWIG C++ processing (default: False)",
+            default = False,
+        ),
+        "extra_args": attr.string_list(
+            doc = "Extra arguments to be passed to SWIG",
+        ),
+        "_swig": attr.label(
+            default = Label("@swig//:swig"),
+            allow_single_file = True,
+            executable = True,
+            cfg = "host",
+        ),
+    },
+)
+
+def swig_go(name, lib, package_name, shared_lib_name=None, tags=[], **kwargs):
+    swig_wrapper_name = name + "_swig"
+
+    _swig_go_wrapper(
+        name = swig_wrapper_name,
+        lib = lib,
+        package_name=package_name,
+        **kwargs,
+    )
+
+    if not shared_lib_name:
+        shared_lib_name = name
+
+    native.filegroup(
+        name = "source_file",
+        srcs = [swig_wrapper_name],
+        output_group = "go_source",
+    )
+
+    # TODO windows would use copy, maybe use an if statement or select to check
```

Comment:
Why don't we fix it right away? It's [cmd_bat](https://bazel.build/reference/be/general#genrule.cmd_bat) for Windows just like you did for Linux.